components/prism-javascript.js

```diff
@@ -9,6 +9,28 @@ Prism.languages.javascript = Prism.languages.extend('clike', {
 Prism.languages.insertBefore('javascript', 'keyword', {
 	'regex': {
 		pattern: /((?:^|[^$\w\xA0-\uFFFF."'\])\s])\s*)\/(\[[^\]\r\n]+]|\\.|[^/\\\[\r\n])+\/[gimyu]{0,5}(?=\s*($|[\r\n,.;})\]]))/,
+		inside: {
+			'regex-flags': {
+				pattern: /[gimyu]{1,5}$/,
+				alias: 'keyword'
+			},
+			'regex-charset': {
+				pattern: /\[(?:\\.|[^\\\]])*\]/,
+				alias: 'escape'
+			},
+			'regex-escape': {
+				pattern: /\\(x[0-9a-f]{2}|u[0-9a-f]{4}|u\{\d+\}|0[0-7]{1,2}|c[a-z]|.)/i,
+				alias: 'escape'
+			},
+			'regex-quantifiers': {
+				pattern: /\{\d+(,\d+)?\}/,
+				alias: 'operator'
+			},
+			'regex-token': {
+				pattern: /[|^$*+(){}]|(?:\?(?![:!=<]))/,
```

Comment:
Sorry, but I don't quite get what `token` is supposed to match.
It includes alternation `|`, anchors `^$`, quantifiers `+*?{}`, groups `()` and the lazy modifier `?`. Also, the matching for `?` is a little strange: `a?=`, `a*?!` are both valid and in neither `?` is matched.

Also, what about the modifiers of groups? E.g. `(?:)`, `(?=)`, `(?!)`.